plugins/file-upload/src/index.jsx

```diff
@@ -1,189 +1,166 @@
-        service,
-        titleText,
-        dragClickLabel,
-        dropLabel,
-        cancelButtonLabel,
-    } = message.data._plugin;
-
-    const {
-        dialogStyles,
-        headerStyles,
-        contentStyles,
-        footerStyles,
-        cancelButtonStyles
-    } = getStylesMemo(theme);
-
-    return (
-        <div
-            {...attributes}
-            style={{
-                ...attributes.styles,
-                ...dialogStyles
-            }}
-        >
-            {(service === 'amazon-s3' || service === 'azure' || service === 'live-agent') &&
-                (!isUploading && !pluginBusy ? (
-                    <>
-                        <header style={headerStyles}>{titleText || 'File Upload'}</header>
-                        <main style={contentStyles} {...getRootProps()}>
-                            <input {...getInputProps()} />
-                            {isDragActive ? (
-                                <p>{dragClickLabel || 'Drop the file here'}</p>
-                            ) : (
-                                    <p>{dropLabel || 'Drag a file here, or click to select one'}</p>
-                                )}
-                            {isRejected ? (
-                                <SizeLimitError />
-                            ) : null}
-                        </main>
-                        <footer style={footerStyles}>
-                            <button
-                                type='button'
-                                onClick={onDismissFullscreen}
-                                style={cancelButtonStyles}
-                            >
-                                {cancelButtonLabel || 'cancel'}
-                            </button>
-                        </footer>
-                    </>
-                ) : (
-                        <>
-                            <header style={headerStyles}>{titleText || 'File Upload'}</header>
-                            <Spinner theme={theme} />
-                        </>
-                    ))}
-        </div>
-    );
+	const [isUploading, setIsUploading] = React.useState(false);
+	const [isRejected, setIsRejected] = React.useState(false);
+
+	const { isFullscreen, onSetFullscreen, theme, message } = props;
+
+	const { openButtonLabel, sizeLimit, fileSizeLimitErrorMessage } = message.data._plugin;
+	// handles change events for the file input
+	// starts the upload
+	const onDrop = React.useCallback(async files => {
+		const file = files[0];
+
+		if (file.size > sizeLimit * 1000 * 1000) {
+			setIsRejected(true);
+			return;
+		}
+
+		setIsRejected(false);
+		try {
+			setIsUploading(true);
+			pluginBusy = true;
+			const result = await upload(props.message.data._plugin, file);
+			setIsUploading(false);
+			pluginBusy = false;
+			if (result.success) {
+				props.onSendMessage("", {
+					file: result.url,
+				});
+				props.onSendMessage("File upload succeeded", {});
+			} else {
+				props.onSendMessage(
```

Comment:
The `sendMessage` api is not a good way to report this error! this will be received and processed as a regular user input by Cognigy.AI.

We should rather set some kind of state, remembering that this particular file upload failed and change the appearance of the "upload" message to indicate the error that happened.